packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -4844,19 +4878,24 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
       _debugLocked = true;
       return true;
     }());
-    final _RouteEntry entry = _history.lastWhere(_RouteEntry.isPresentPredicate);
-    if (entry.hasPage) {
-      if (widget.onPopPage!(entry.route, result))
+    final int index = _history.lastIndexWhere(_RouteEntry.isPresentPredicate);
```

Comment:
Why the change from lastWhere to lastIndexWhere? Looks like you need to look up the entry anyways and don't need the index really?

Reply:
oops it looks like i forgot to clean this up